components/style/stylist.rs

```diff
@@ -375,10 +376,7 @@ impl Stylist {
     #[inline]
     fn note_for_revalidation(&mut self, selector: &Selector<SelectorImpl>) {
         if needs_revalidation(selector) {
-            // For revalidation, we can skip everything left of the first ancestor
-            // combinator.
-            let revalidation_sel = selector.inner.slice_to_first_ancestor_combinator();
```

Comment:
We should probably get rid of the slice_to_first_ancestor_combinator() API while we're at it. I don't think we use it for anything else.